tests/test_schema.py

```diff
@@ -0,0 +1,244 @@
+    with pytest.raises(dj.errors.AccessError):
+        info_schema.drop()
+
+
+def test_namespace_population(schema_empty_module):
+    """
+    With the schema_empty_module fixture, this test
+    mimics the behavior of `spawn_missing_classes`, as if the schema
+    was declared in a separate module and `spawn_missing_classes` was called in that namespace.
+    """
+    # Spawn missing classes in the caller's (self) namespace.
+    schema_empty_module.schema.context = None
+    schema_empty_module.schema.spawn_missing_classes(context=None)
+    # Then add them to the mock module's namespace.
+    for k, v in locals().items():
+        if inspect.isclass(v):
+            setattr(schema_empty_module, k, v)
+
+    for name, rel in getmembers(schema, relation_selector):
+        assert hasattr(schema_empty_module, name), "{name} not found in schema_empty".format(name=name)
+        assert rel.__base__ is getattr(schema_empty_module, name).__base__, "Wrong tier for {name}".format(name=name)
+
+        for name_part in dir(rel):
+            if name_part[0].isupper() and part_selector(getattr(rel, name_part)):
+                assert getattr(rel, name_part).__base__ is dj.Part, "Wrong tier for {name}".format(name=name_part)
+
+
+def test_undecorated_table():
+    """
+    Undecorated user table classes should raise an informative exception upon first use
+    """
+
+    class UndecoratedClass(dj.Manual):
+        definition = ""
+
+    a = UndecoratedClass()
+    with pytest.raises(dj.DataJointError):
+        print(a.full_table_name)
+
+
+def test_reject_decorated_part(schema_any):
+    """
+    Decorating a dj.Part table should raise an informative exception.
+    """
+
+    class A(dj.Manual):
+        definition = ...
+
+        class B(dj.Part):
+            definition = ...
+
+
+    with pytest.raises(dj.DataJointError):
+        schema_any(A.B)
+        schema_any(A)
+
+
+def test_unauthorized_database(db_creds_test):
+    """
+    an attempt to create a database to which user has no privileges should raise an informative exception.
+    """
+    with pytest.raises(dj.DataJointError):
+        dj.Schema("unauthorized_schema", connection=dj.conn(reset=True, **db_creds_test))
+
+
+def test_drop_database(db_creds_test):
+    schema = dj.Schema(
+        PREFIX + "_drop_test", connection=dj.conn(reset=True, **db_creds_test)
+    )
+    assert schema.exists
+    schema.drop()
+    assert not schema.exists
+    schema.drop()  # should do nothing
+
+
+def test_overlapping_name(connection_test):
+    test_schema = dj.Schema(
+        PREFIX + "_overlapping_schema", connection=connection_test
+    )
+
+    @test_schema
+    class Unit(dj.Manual):
+        definition = """
+        id:  int     # simple id
+        """
+
+    # hack to update the locals dictionary
+    locals()
+
+    @test_schema
+    class Cell(dj.Manual):
+        definition = """
+        type:  varchar(32)    # type of cell
+        """
+
```

Comment:
A bit inconsistent in the way we add `Cell` to `test_schema`, but this is exactly the way it was written in the old tests, so I'm just going to leave it.